src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java

```diff
@@ -639,13 +645,13 @@ private static GitHubConnection connect(@NonNull ConnectionId connectionId, @Non
         }
 
 
-        private void release() {
-            if (this.usageCount <= 1) {
-                this.usageCount = 0;
-                this.lastUsed = System.currentTimeMillis();
-            } else {
-                this.usageCount -= 1;
+        private void release() throws IOException {
+            if (this.usageCount <= 0) {
+                throw new IOException("Tried to release a GitHubConnection that should have no references.");
```

Comment:
I totally agree with this change, but might be good to test it on a real instance for a few days in case there are any issues not caught by the test suite.

Reply:
it's running on ci.jenkins.io now